agent/src/main/java/com/teamscale/jacoco/agent/options/AgentOptions.java

```diff
@@ -495,8 +496,12 @@ private IUploader createTeamscaleSingleProjectUploader(Instrumentation instrumen
 	private DelayedTeamscaleMultiProjectUploader createTeamscaleMultiProjectUploader(
 			Instrumentation instrumentation) {
 		DelayedTeamscaleMultiProjectUploader uploader = new DelayedTeamscaleMultiProjectUploader(
-				(project, revision) ->
-						new TeamscaleUploader(teamscaleServer.withProjectAndRevision(project, revision)));
+				(project, commitInfo) -> {
+					if (commitInfo.preferCommitDescriptorOverRevision) {
```

Comment:
I don't fully understand. What is your suggestion? We somehow need to figure out, if we set `commitInfo.commit` via the `teamscale.*` or the `git.*` properties. That's what this flag is about. 
Ofc, we could also just not parse the `git.commit.branch` and `git.commit.time` properties and check for `commitInfo.commit != null` here, but this just makes the code less readable because the upload preference is quite implicit then. This way, the code is more readable, imo.

Reply:
Ah, I see. That's a very good point, then :D